src/sentry/api/endpoints/project_stacktrace_link.py

```diff
@@ -273,28 +313,75 @@ def get(self, request: Request, project: Project) -> Response:
                     )
                     and project.organization.flags.codecov_access
                 )
+
                 # Check the cache and skip querying Codecov if a query within the last hour found no integration.
                 codecov_cache = cache.get(cache_key.format(project_id=project.id))
                 if codecov_enabled and codecov_cache is False:
                     result["codecov"] = {"status": 404}
 
                 elif codecov_enabled and codecov_cache is not False:
+                    # Get Codecov data
                     try:
-                        lineCoverage, codecovUrl = get_codecov_data(
-                            repo=current_config["config"]["repoName"],
-                            service=current_config["config"]["provider"]["key"],
-                            branch=current_config["config"]["defaultBranch"],
-                            path=current_config["outcome"]["sourcePath"],
+                        ref = (
+                            ctx["commit_id"]
+                            if ctx["commit_id"]
+                            else current_config["config"]["defaultBranch"]
                         )
-                        if lineCoverage and codecovUrl:
-                            result["codecov"] = {
-                                "lineCoverage": lineCoverage,
-                                "coverageUrl": codecovUrl,
-                                "status": 200,
-                            }
-
-                            if not codecov_cache:
-                                cache.set(cache_key.format(project_id=project.id), True, 3600)
+
+                        # Get commit sha from Git blame
+                        codecov_commit_sha_from_git_blame_enabled = features.has(
```

Comment:
we could shorten this variable name and make it more readable with `use_commit_sha`. As someone reading the code months later, I'm not necessarily concerned with where we're getting the SHA from.

Reply:
Maybe `fetch_commit_sha`?

Reply:
sounds great! `should_` or `enable_` can be great prefixes to indicate a variable is a boolean gate for an action too